sqlite.zig

```diff
@@ -1452,8 +1452,8 @@ pub const DynamicStatement = struct {
     ///
     /// Possible errors:
     /// - SQLiteError.SQLiteNotFound if some fields not found
-    pub fn iterator(self: *Self, comptime Type: type, values: anytype) !Iterator(Type) {
-        try self.smartBind(values);
+    pub fn iterator(self: *Self, comptime Type: type, options: anytype, values: anytype) !Iterator(Type) {
```

Comment:
I would prefer we add `iteratorAlloc`  like [here](https://github.com/vrischmann/zig-sqlite/blob/master/sqlite.zig#L1724-L1733). We stopped exposing `options: anytype` to the user and instead clearly separate the allocation-capable function from the other.

Reply:
Sure, i'll change it 🙂

Reply:
Added a second `iteratorAlloc` function